### ISSUE TITLE
'err' object is not passed into callback when response has a non-200 code but also has valid JSON

### ISSUE THREAD

**Issue by jcfrancisco** [groupon/gofer#34]
https://github.com/groupon/gofer/blob/40731c12f65a6ae68913ef4768e3906083547776/src/gofer.coffee#L177

```
    if typeof cb == 'function'
      @hub.fetch options, (error, body, response, responseData) ->
        parseJSON = options.parseJSON ? isJsonResponse(response, body)
        {error, body} = safeParseJSON body, response if parseJSON
        # TODO: remove flipping of response and responseData with next major
        cb error, body, responseData, response
```

on a 500 response w/ valid JSON, the first `error` gets blown away, and `error` gets set to null in safeParseJson (because the JSON is valid).

**Comment by jcfrancisco** [comment]
Test case we noticed this in:

error code: `500 Internal Error`
body: `{"error": "Error: API Request returned a response outside the status code range (code: 500, range: [200, 299])"}`

**Comment by jkrems** [comment]
Oops, that sounds pretty serious. Thanks!

**Comment by jkrems** [comment]
Should be fixed in v2.3.4
